src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1517,8 +1517,16 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
 
         assert((simdSize == 8) || (simdSize == 12) || (simdSize == 16) || (simdSize == 32));
 
-        if ((argCnt == 1) ||
-            ((vecCns.i64[0] == vecCns.i64[1]) && ((simdSize <= 16) || (vecCns.i64[2] == vecCns.i64[3]))))
+        bool i64ComponentsEqual = true;
+        for (UINT32 i = 0; i < simdSize / 8; i++)
```

Comment:
There's actually a small bug here. `simdSize / 8` will be `1` for `simdSize == 12` and so it will miss the upper 4 bytes.
Likewise, doing a full secondary 8-byte comparison will fail for `simdSize == 12` because the unused bytes 12-15 will always be zero.

The general intent of the first group is to catch the case where all elements are functionally equal and therefore replacing it with `get_Zero` or `get_AllBitsSet` is possible and so future replacements can likewise take place, if appropriate (e.g. it might be better to optimize to be `broadcast` or `shuffle` in some cases, like `Clang` does).

In practice, it handles this for `argCnt == 1` by ensuring all elements are filled in here: https://github.com/dotnet/runtime/pull/47385/files#diff-8fe1c38ca575e327ad3227c06d009135287a438045cecbe3301d9a633a46cf0cR1465

However, I think it misses the mark when the args are explicitly defined or when `simdSize == 12` and so the logic here likely needs adjusting.
The bytewise comparison that was being done before also misses the mark on checking "are all elements the same"

Reply:
Given this probably needs to be backported, it might be good to have a simple fix there and to have a slightly more complex fix for .NET 6+?

In particular, the logic to determine "are all elements equal?" could be put into the `HandleArgForHWIntrinsicCreate` logic here: https://github.com/dotnet/runtime/pull/47385/files#diff-8fe1c38ca575e327ad3227c06d009135287a438045cecbe3301d9a633a46cf0cR1430-R1470, so we aren't looping or checking values unnecessarily.

We could then trivially check:
```
if (allElementsEqual)
{
    if (allBitsZero)
    {
    }
    else if (allBitsSet)
    {
    }
    // future logic handling other cases
}
```

Thoughts?